packages/cli/src/fulltext/listProjectFiles.ts

```diff
@@ -20,17 +20,17 @@ export default async function listProjectFiles(
   // Perform a breadth-first traversal of a directory, collecting all non-binary files and
   // applying the directory ignore list.
   const processDir = async (dir: string) => {
-    const queue = [dir];
+    const queue = ['.'];
     while (queue.length > 0 && files.length < fileLimit) {
       const currentDir = queue.shift();
       assert(currentDir, 'queue should not be empty');
 
-      const entries = await readdir(currentDir, { withFileTypes: true });
+      const entries = await readdir(join(dir, currentDir), { withFileTypes: true });
       for (const entry of entries) {
-        const fullPath = join(currentDir, entry.name);
+        const path = currentDir === '.' ? entry.name : join(currentDir, entry.name);
         if (entry.isDirectory()) {
-          if (!ignoreDirectory(entry.name)) queue.push(fullPath);
-        } else files.push(fullPath);
+          if (!ignoreDirectory(entry.name)) queue.push(path);
+        } else files.push(path);
       }
     }
```

Comment:
nit - Using `relative` might better capture the intention of the path transformations here
```ts
  const processDir = async (dir: string) => {
    const queue = [dir];
    while (queue.length > 0 && files.length < fileLimit) {
      const currentDir = queue.shift();
      assert(currentDir, 'queue should not be empty');

      const entries = await readdir(currentDir, { withFileTypes: true });
      for (const entry of entries) {
        const path = join(currentDir, entry.name);
        if (entry.isDirectory()) {
          if (!ignoreDirectory(entry.name)) queue.push(path);
        } else {
          files.push(relative(directory, path));
        }
      }
    }
  };
  ```

Reply:
OK, I'll change in a different branch.